e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,529 @@
+	expectedGetMasterMinterResponse := fiattokenfactorytypes.QueryGetMasterMinterResponse{
+		MasterMinter: fiattokenfactorytypes.MasterMinter{
+			Address: string(newMM1.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Update Master Minter from non owner account
+	// EXPECTED: Request fails; Master Minter not updated
+	// Status:
+	// 	Master Minter: newMM1
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble)
+	newMM2 := w[0]
+	alice := w[1]
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "update-master-minter", newMM2.FormattedAddress())
+	require.ErrorContains(t, err, "you are not the owner: unauthorized")
+
+	showMMRes, err = showMasterMinter(ctx, val)
+	require.NoError(t, err, "failed to query show-master-minter")
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+
+	// ACTION: Update Master Minter from blacklisted owner account
+	// EXPECTED: Success; Master Minter updated
+	// Status:
+	// 	Master Minter: newMM1
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-master-minter", newMM2.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-master-minter message")
+
+	showMMRes, err = showMasterMinter(ctx, val)
+	require.NoError(t, err, "failed to query show-master-minter")
+	expectedGetMasterMinterResponse = fiattokenfactorytypes.QueryGetMasterMinterResponse{
+		MasterMinter: fiattokenfactorytypes.MasterMinter{
+			Address: string(newMM2.FormattedAddress()),
+		},
+	}
+
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	// ACTION: Update Master Minter to blacklisted Master Minter account
+	// EXPECTED: Success; Master Minter updated
+	// Status:
+	// 	Master Minter: newMM2
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "new-mm-3", math.OneInt(), noble)
+	newMM3 := w[0]
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, newMM3)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-master-minter", newMM3.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-master-minter message")
+
+	showMMRes, err = showMasterMinter(ctx, val)
+	require.NoError(t, err, "failed to query show-master-minter")
+	expectedGetMasterMinterResponse = fiattokenfactorytypes.QueryGetMasterMinterResponse{
+		MasterMinter: fiattokenfactorytypes.MasterMinter{
+			Address: string(newMM3.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetMasterMinterResponse.MasterMinter, showMMRes.MasterMinter)
+}
+
+func TestFiatTFConfigureMinterController(t *testing.T) {
```

Comment:
missing the happy path test here too :)

Reply:
Added in: https://github.com/noble-assets/noble/pull/368/commits/a278e1e59be0319c50ef61a3a1ff79105c8fa702